database/staking.go

```diff
@@ -439,3 +447,37 @@ func (db BigDipperDb) SaveRedelegations(redelegations []types.Redelegation) erro
 	_, err = db.Sql.Exec(redelQuery, redelParams...)
 	return err
 }
+
+//SaveDelegationsShare sve an array of delegation share
+func (db BigDipperDb) SaveDelegationsShares(shares []types.DelegationShare) error {
+	stmt := `INSERT INTO validator_delegation_shares (operator_address ,delegator_address,shares,height,timestamp) VALUES`
+	var delegationShareParam []interface{}
+	for i, share := range shares {
+		i1 := i * 5
+		stmt += fmt.Sprintf("($%d,$%d,$%d,$%d,$%d),", i1+1, i1+2, i1+3, i1+4, i1+5)
+		delegationShareParam = append(delegationShareParam, share.ValidatorAddress.String(), share.DelegatorAddress.String(), share.Shares, share.Height, share.Timestamp)
+	}
+	stmt = stmt[:len(stmt)-1] // Remove the trailing ","
+	stmt += " ON CONFLICT DO NOTHING"
+	_, err := db.Sql.Exec(stmt, delegationShareParam...)
+	if err != nil {
+		return err
+	}
+
+	//insert self delegation in this entry into self_deleation_row
+	//for each time there are someone delegated to that validator, save the validator_self delegation?
```

Comment:
Were you able to answer this question?

Reply:
Oh I just forget to delete the question mark😂